.azure-pipelines/ultimate-pipeline.yml

```diff
@@ -1105,6 +1108,205 @@ stages:
+
+    - powershell: |
+        # Prepare environment
+        $env:DD_INSIDE_CI=$true
+        $env:DD_HOSTNAME="dd-profiler-dotnet"
+        $env:DD_SERVICE="dd-profiler-dotnet"
+        $env:DD_PROFILING_ENABLED="1"
+
+        # Set profiler deployment and test output dir folder path for the test
+        $test_output_dir="$(System.DefaultWorkingDirectory)\test_output"
+
+        $env:DD_TESTING_OUPUT_DIR=$test_output_dir
+        $env:DD_TRACE_DEBUG="1"
+        $env:DD_PROFILING_LOG_DIR="$test_output_dir\logs"
+        $env:DD_PROFILING_OUTPUT_DIR="$test_output_dir\pprofs"
+        $env:DD_TRACE_ENABLED="0"
+        $env:COR_ENABLE_PROFILING="1"
+        $env:COR_PROFILER="{846F5F1C-F9AE-4B07-969E-05C26BC060D8}"
+        $env:COR_PROFILER_PATH="$(Join-Path -Path $(Resolve-Path .) -ChildPath shared/bin/monitoring-home/win-$(platform)/Datadog.Trace.ClrProfiler.Native.dll)"
+        $env:DD_DOTNET_PROFILER_HOME="$(Join-Path -Path $(Resolve-Path .) -ChildPath shared/bin/monitoring-home)"
+
+        # Run Computer01 for 2min
+        cd profiler\_build\bin\Release-$(platform)\profiler\src\Demos\Samples.Computer01\net45
+        .\Samples.Computer01.exe --scenario 1 --timeout 120
+      displayName: run asan test
+      workingDirectory: $(System.DefaultWorkingDirectory)
+
+    - publish: $(System.DefaultWorkingDirectory)/test_output
+      displayName: Uploading test results
+      artifact: asan_tests_windows_profiler_$(platform)_$(System.JobAttempt)
+      condition: succeededOrFailed()
+      continueOnError: false
+
+- stage: static_analysis_tests_windows_profiler
+  condition: > 
+    and(
+      succeeded(), 
+      eq(variables['isBenchmarksOnlyBuild'], 'False'),
+      eq(dependencies.generate_variables.outputs['generate_variables_job.generate_variables_step.IsProfilerChanged'], 'True')
+    )
+  dependsOn: [master_commit_id, generate_variables, build_windows_tracer, build_windows_profiler]
+  variables:
+    masterCommitId: $[ stageDependencies.master_commit_id.fetch.outputs['set_sha.master']]
+  jobs:
+  - template: steps/update-github-status-jobs.yml
+    parameters:
+      jobs: [Win]
+
+  - job: Win
+    pool:
+      vmImage: windows-2019
+    timeoutInMinutes: 60 #default value
+    strategy:
+      matrix:
+        x86:
+          platform: x86
+        x64:
+          platform: x64
+    variables:
+      USE_NATIVE_LOADER: true
+    
+    steps:
+    - template: steps/install-dotnet-sdks.yml
+
+    - task: DownloadPipelineArtifact@2
+      displayName: Download windows profiler home
+      inputs:
+        artifact: windows-profiler-binaries
+        path: $(System.DefaultWorkingDirectory)/profiler/_build
+
+    - template: steps/restore-working-directory.yml
+      parameters:
+        artifact: windows-tracer-home
+
+    - powershell: |
+        choco install cppcheck -y
+        Write-Host "##vso[task.setvariable variable=PATH;]${env:PATH};C:\Program Files\Cppcheck";
+        refreshenv
+      displayName: Install CppCheck
```

Comment:
Currently we're running cppcheck against the tracer and the native loader in the BuildTracer stage (Using Nuke I might add 😛).

1. Maybe we should stop running in that stage and move all the checking to a separate stage
2. You don't need to install cppcheck (we're using it in the build stage already, so it's clearly there)
3. This would all be a lot easier with Nuke 😛

Reply:
Follow up task? 🙏

Reply:
From what I see, CppCheck is run on linux for the tracer and native loader not on windows.
Maybe that's relevant for the Tracer but for the profiler with have a bunch of specific Windows code.